com.unity.ml-agents/Runtime/Sensors/ObservationWriter.cs

```diff
@@ -38,6 +38,10 @@ internal void SetTarget(IList<float> data, int[] shape, int offset)
             {
                 m_TensorShape = new TensorShape(m_Batch, shape[0]);
             }
+            else if (shape.Length == 2)
+            {
```

Comment:
You're probably going to need to implement `public float this[int, int]` at some point to actual set the data, but this is fine for now.